tm2/pkg/telemetry/config/config.go

```diff
@@ -8,11 +8,14 @@ var errEndpointNotSet = errors.New("telemetry exporter endpoint not set")
 
 // Config is the configuration struct for the tm2 telemetry package
 type Config struct {
-	MetricsEnabled    bool   `json:"enabled" toml:"enabled"`
+	MetricsEnabled    bool   `json:"metrics_enabled" toml:"metrics_enabled"`
 	MeterName         string `json:"meter_name" toml:"meter_name"`
 	ServiceName       string `json:"service_name" toml:"service_name" comment:"in Prometheus this is transformed into the label 'exported_job'"`
 	ServiceInstanceID string `json:"service_instance_id" toml:"service_instance_id" comment:"the ID helps to distinguish instances of the same service that exist at the same time (e.g. instances of a horizontally scaled service), in Prometheus this is transformed into the label 'exported_instance"`
 	ExporterEndpoint  string `json:"exporter_endpoint" toml:"exporter_endpoint" comment:"the endpoint to export metrics to, like a local OpenTelemetry collector"`
+
+	TracesEnabled             bool `json:"traces_enabled" toml:"traces_enabled"`
+	GracefulShutdownTelemetry bool `json:"graceful_shutdown_telemetry" toml:"graceful_shutdown_telemetry"`
```

Comment:
Why would we ever not want to gracefully shut down telemetry?

Reply:
Originally there was no Graceful shutdown, so I added this for case where we need that. I think that we will need GracefulShutdown when we run it in production and does not need it when running locally or test. Wdyt ?